keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourceActuator.kt

```diff
@@ -121,9 +121,14 @@ class ResourceActuator(
 
                   artifactRepository.markAsVetoedIn(
                     deliveryConfig = deliveryConfig,
-                    artifact = artifact,
-                    version = artifactVersion,
-                    targetEnvironment = environment)
+                    veto = EnvironmentArtifactVeto(
+                      reference = artifact.reference,
+                      version = artifactVersion,
+                      targetEnvironment = environment,
+                      vetoedBy = "Spinnaker",
+                      comment = "Automatically vetoed because deployment of this version failed."
+                    )
```

Comment:
This creates an automated user + reason for automatic rollbacks. Thoughts on wording here?

Reply:
Maybe "this version is vetoed automatically as we tried to deploy it multiple times and it failed"?

Reply:
I like @gal-yardeni's idea of communicating that there were multiple attempts.

Reply:
Updated to indicate that there were multiple deployments